subworkflows/local/singular_demultiplex/main.nf

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env nextflow
+
+//
+// Demultiplex Singular Genomics bases data using sgdemux
+//
+
+include { SGDEMUX } from "../../../modules/nf-core/sgdemux/main"
+
+workflow SINGULAR_DEMULTIPLEX {
+    take:
+        // Input fastq's must be bgzipped for compatibility with sgdemux
+        // samplesheet.csv must be a two column csv = Sample_Barcode,Sample_ID)
+        ch_flowcell     // [[id:"", lane:""],samplesheet.csv, path/to/bases/files]
+
+    main:
+
+        // MODULE: sgdemux
+        // Demultiplex the bases files
+        SGDEMUX( ch_flowcell )
+
+        // Generate meta for each fastq
+        ch_fastq_with_meta = generate_fastq_meta(SGDEMUX.out.sample_fastq)
+
+    emit:
+        fastq                   = ch_fastq_with_meta
+        // sample_json             = BASES2FASTQ.out.sample_json TODO no json
+        // qc_report               = BASES2FASTQ.out.qc_report TODO no qc report
+        generated_run_manifest  = SGDEMUX.out.per_project_metrics // TODO
+        metrics                 = SGDEMUX.out.metrics // TODO 
+        unassigned              = SGDEMUX.out.most_frequent_unmatched // TODO 
+        versions                = SGDEMUX.out.versions
+}
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    FUNCTIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// Add meta values to fastq channel
+def generate_fastq_meta(ch_reads) {
```

Comment:
This looks very similar to https://github.com/fulcrumgenomics/nfcore-demultiplex/blob/8e42b6cb4e12bf00828fa8cb282e01bcb487fe6e/subworkflows/local/bases_demultiplex/main.nf#L38-L67.  So why not just refactor this by moving it to `./lib` (https://github.com/fulcrumgenomics/nfcore-demultiplex/tree/master/lib) and then using it in both places?